collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -693,6 +658,51 @@ private void loadFromIntent(Intent intent) {
         formLoaderTask.execute(formPath);
     }
 
+    private String loadSavePoint() {
+        final String filePrefix = formPath.substring(
+                formPath.lastIndexOf('/') + 1,
+                formPath.lastIndexOf('.'))
+                + "_";
+        final String fileSuffix = ".xml.save";
+        File cacheDir = new File(storagePathProvider.getOdkDirPath(StorageSubdirectory.CACHE));
+        File[] files = cacheDir.listFiles(pathname -> {
+            String name = pathname.getName();
+            return name.startsWith(filePrefix)
+                    && name.endsWith(fileSuffix);
+        });
+
+        if (files != null) {
+            /**
+             * See if any of these savepoints are for a filled-in form that has never
+             * been explicitly saved by the user.
+             */
+            for (File candidate : files) {
+                String instanceDirName = candidate.getName()
+                        .substring(
+                                0,
+                                candidate.getName().length()
+                                        - fileSuffix.length());
+                File instanceDir = new File(
+                        storagePathProvider.getOdkDirPath(StorageSubdirectory.INSTANCES) + File.separator
+                                + instanceDirName);
+                File instanceFile = new File(instanceDir,
+                        instanceDirName + ".xml");
+                if (instanceDir.exists()
+                        && instanceDir.isDirectory()
+                        && !instanceFile.exists()) {
+                    // yes! -- use this savepoint file
+                    return instanceFile
+                            .getAbsolutePath();
+                }
+            }
+
+        } else {
+            Timber.e("Couldn't access cache directory when looking for save points!");
```

Comment:
This will just mean we get a crash when trying to save a savepoint, right? Should we preemptively address that?

Reply:
Oh I can check. All our tests actually pass in the scenario where the cache directory is never created, so it might actually be fine - I was thinking it was tested, but it might not be.

Reply:
I tested savepoints without a cache directory and nothing exploded. If you look at `SavePointTask` and the call to `writeFile` you'll see that we create the directory if needed, which must be why. That's nice as it's closer to the "lazy" creation of directories I'd like to see us move towards anyway.

Reply:
Oh, how fabulous! Thanks for the extra check.